flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterJobClient.java

```diff
@@ -43,23 +43,30 @@
 /**
  * A {@link JobClient} for a {@link MiniCluster}.
  */
-public final class PerJobMiniClusterJobClient implements JobClient, CoordinationRequestGateway {
+public final class MiniClusterJobClient implements JobClient, CoordinationRequestGateway {
 
-	private static final Logger LOG = LoggerFactory.getLogger(PerJobMiniClusterJobClient.class);
+	private static final Logger LOG = LoggerFactory.getLogger(MiniClusterJobClient.class);
 
 	private final JobID jobID;
 	private final MiniCluster miniCluster;
-	private final CompletableFuture<JobResult> jobResultFuture;
 	private final ClassLoader classLoader;
+	private final CompletableFuture<JobResult> jobResultFuture;
 
-	public PerJobMiniClusterJobClient(JobID jobID, MiniCluster miniCluster, ClassLoader classLoader) {
+	/**
+	 * Creates a {@link MiniClusterJobClient} for the given {@link JobID} and {@link MiniCluster}.
+	 * This will shut down the {@code MiniCluster} after job result retrieval if {@code
+	 * shutdownCluster} is {@code true}.
+	 */
+	public MiniClusterJobClient(JobID jobID, MiniCluster miniCluster, ClassLoader classLoader, boolean shutdownCluster) {
```

Comment:
I tend to prefer enums instead of booleans for parameters because it makes it explicit what `true` and `false` means.